blocks/Xcos/blocks/CLOCK_c.py

```diff
@@ -42,55 +53,55 @@ def CLOCK_c(outroot, attribid, ordering, geometry, parameters):
+                  parent=block_id[0], a='')
 
     # Create the EventOutBlock node inside the root tag
-    CLKOUT_f(root, attribid, ordering, geometry, parameters)
+    CLKOUT_f(root, block_id[1], ordering, geometry, parameters)
 
-    addPort(root, TYPE_CNTRL, id="-63efee48:189fd5ed04e:-73dd",
-            parent="-73e75f0:167968eb73f:-7955", ordering="1",
+    addPort(root, TYPE_CNTRL, id=port_id[0],
+            parent=block_id[1], ordering="1",
             dataType="REAL_MATRIX", dataColumns="1",
             dataLines="-1", initialState="0.0",
             style="ControlPort", value="")
 
-    EVTDLY_c(root, attribid, ordering, geometry, parameters)
+    EVTDLY_c(root, block_id[2], ordering, geometry, parameters)
 
-    addPort(root, TYPE_CNTRL, id="-63efee48:189fd5ed04e:-73db",
-            parent="-73e75f0:167968eb73f:-7953", ordering="1",
+    addPort(root, TYPE_CNTRL, id=port_id[1],
+            parent=block_id[2], ordering="1",
             dataType="REAL_MATRIX", dataColumns="1",
             dataLines="-1", initialState="0.0",
             style="ControlPort", value="")
-    addPort(root, TYPE_CMD, id="-63efee48:189fd5ed04e:-73da",
-            parent="-73e75f0:167968eb73f:-7953", ordering="1",
+    addPort(root, TYPE_CMD, id=port_id[2],
+            parent=block_id[2], ordering="1",
             dataType="REAL_MATRIX", dataColumns="1",
             dataLines="-1", initialState="0.0",
             style="CommandPort", value="")
 
     SplitBlock(root, attribid, ordering, geometry)
 
-    addPort(root, TYPE_CNTRL, id="-63efee48:189fd5ed04e:-73d8",
-            parent="-73e75f0:167968eb73f:-7950", ordering="1",
+    addPort(root, TYPE_CNTRL, id=port_id[3],
+            parent=block_id[3], ordering="1",
             dataType="REAL_MATRIX", dataColumns="1",
             dataLines="-1", initialState="0.0",
             style="ControlPort", value="")
-    addPort(root, TYPE_CMD, id="-63efee48:189fd5ed04e:-73d7",
-            parent="-73e75f0:167968eb73f:-7950", ordering="1",
+    addPort(root, TYPE_CMD, id=port_id[4],
+            parent=block_id[3], ordering="1",
             dataType="REAL_MATRIX", dataColumns="1",
             dataLines="-1", initialState="-1.0",
             style="CommandPort", value="")
-    addPort(root, TYPE_CMD, id="-63efee48:189fd5ed04e:-73d6",
-            parent="-73e75f0:167968eb73f:-7950", ordering="2",
+    addPort(root, TYPE_CMD, id=port_id[5],
+            parent=block_id[3], ordering="2",
             dataType="REAL_MATRIX", dataColumns="1",
             dataLines="-1", initialState="-1.0",
             style="CommandPort", value="")
 
-    CCLink = addLink(root, TYPE_LINK, id="-63efee48:189fd5ed04e:-73d5",
-                     parent="128c18ea:1383ab8277e:-748d",
-                     source="-63efee48:189fd5ed04e:-73d6",
-                     target="-63efee48:189fd5ed04e:-73db",
+    CCLink = addLink(root, TYPE_LINK, id=link_id[0],
+                     parent=link_id[3],
```

Comment:
```suggestion
                     parent=block_id[0],
```
replace with block_id[0]